SpringBootAngularCrud/src/main/java/com/mvc/service/CustomerServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package com.mvc.service;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.mvc.dto.CustomerDto;
+import com.mvc.entity.Customer;
+import com.mvc.repository.CustomerRepository;
+
+@Service
+public class CustomerServiceImpl implements CustomerService {
+
+	@Autowired
+	private CustomerRepository repository;
+
+	@Override
+	public List<Customer> getAllCustomerList() {
+		return repository.findAll();
+	}
+
+	@Override
+	public String addCustomer(CustomerDto customerDto) {
+		boolean isMobileExists = repository.existsByMobileNo(customerDto.getMobileNo());
+		boolean isEmailExists = repository.existsByEmail(customerDto.getEmail());
+		
+		// check if email and mobile number already register
+		if(isMobileExists && isEmailExists) {
+			return "Mobile number and email already registered";
+			
+		} else if(isMobileExists) {   
+			return "Mobile number already registered";
+			
+		} else if(isEmailExists) {
+			return "Email already registered";
+			
+		} else {        // if none of mobile or email register then save the data
+			Customer customer = new Customer();
+			
+			customer.setFirstName(customerDto.getFirstName());
+			customer.setLastName(customerDto.getLastName());
+			customer.setDateOfBirth(customerDto.getDateOfBirth());
+			customer.setMobileNo(customerDto.getMobileNo());
+			customer.setAddressLineOne(customerDto.getAddressLineOne());
+			customer.setAddressLineTwo(customerDto.getAddressLineTwo());
+			customer.setAge(customerDto.getAge());
+			customer.setGender(customerDto.getGender());
+			customer.setEmail(customerDto.getEmail());
+			
+			repository.save(customer);
+			return "Data saved";
+		}
+	}
+
+	@Override
+	public String updateCustomerDetails(long id, CustomerDto updatedCutomerDto) {
```

Comment:
Made a common method for saving and update and also @pathvariable for id. Instead get ID from the Dto